### PR TITLE
Group edit refactor

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by akatsoulas** [comment]
This PR needs some cleaning and an update on the tests. It's basically a draft but since this is a big chunk of code I opened in for a first review.

@johngian a few things that are not ready yet:
* Form errors in case that the data are not correct
* Add by default a curator to a new group
* Correctly add a new group

File: mozillians/groups/forms.py [hunk 1]
```diff
@@ -22,7 +23,64 @@ def clean_sort(self):
...
+            msg = _(u'The group must have at least one curator.')
+            self._errors['curators'] = self.error_class([msg])
+
+        return cleaned_data
+
+    def save(self, *args, **kwargs):
+        """Custom save method to add multiple curators."""
+        obj = super(GroupCuratorsForm, self).save(*args, **kwargs)
+
+        for curator in self.cleaned_data['curators']:
```
Comment (johngian) [inline mozillians/groups/forms.py]: You should also make sure here you delete any previous curators not in current request.

File: mozillians/jinja2/groups/includes/group_edit_invitation.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<div role="tabpanel" class="tab-pane fade" id="invitation">
+  <form class="edit-group" method="POST" action="{{ url('groups:group_edit', group.url) }}?next=invitation">
+    {% csrf_token %}
+    <div class="panel panel-default">
+      <div class="panel-heading">{{ _('Pending Invitations') }}</div>
+      <div class="panel-body">
+
+        <fieldset>
+          {% for redeemer in invites %}
```
Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: Invites don't always exist

File: mozillians/groups/views.py [hunk 3]
```diff
@@ -394,32 +395,58 @@ def group_add_edit(request, url=None):
...
-    form_class = SuperuserGroupForm if is_manager else GroupForm
-
-    # Add the creator of a group as curator
-    curators_ids = [profile.id]
-    # If we are editing add the existing curators. If the group has no curator in edit
-    # mode, append an empty list
-    if url:
-        curators_ids = group.curators.all().values_list('id', flat=True)
+        for k in group_forms.keys():
+            if not k == key:
```
Comment (johngian) [inline mozillians/groups/views.py]: Please squash the functionality here with the line above.

File: mozillians/jinja2/groups/includes/group_edit_description.html [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  {% if show_delete_group_button %}
+    <div class="panel panel-danger">
+      <div class="panel-heading">
+        {{ _('Delete Group') }}
+      </div>
+      <div class="panel-body">
+        <fieldset id="delete">
+          <input id="delete-checkbox" type="checkbox" class="acknowledge">
+          <label for="delete-checkbox">
+            {{ _('I acknowledge that deleting this group') }}
```
Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_description.html]: Fix text here.

File: mozillians/groups/forms.py [hunk 5]
```diff
@@ -31,14 +88,108 @@ class GroupForm(happyforms.ModelForm):
...
+        super(GroupInviteForm, self).__init__(*args, **kwargs)
+        self.fields['invites'].required = False
+        self.fields['invites'].help_text = _('Start typing the name/email/username '
+                                             'of a vouched Mozillian.')
+
+    def clean(self):
+        """Custom clean method."""
+        super(GroupInviteForm, self).clean()
+        user = self.request.user
+        if not user.is_superuser or user.userprofile not in self.instance.curators.all():
```
Comment (johngian) [inline mozillians/groups/forms.py]: Please filter and check instead if testing existence in qs.

File: mozillians/groups/forms.py [hunk 6]
```diff
@@ -31,14 +88,108 @@ class GroupForm(happyforms.ModelForm):
...
+            msg = _(u'You need to be the curator of this group before inviting someone to join.')
+            self._errors['invites'] = self.error_class([msg])
+            del self.cleaned_data['invites']
```
Comment (johngian) [inline mozillians/groups/forms.py]: Why do we need this ``del``.

**Comment by johngian** [comment]
Tested it locally, and there are many things that need improvement. Basic functionality looks OK. Let's tackle the failing tests in an other PR.

**Comment by akatsoulas** [comment]
Although this is not polished, I am merging for demo purposes.

**Pull request merged**